### PR TITLE
Update of ProjectStructureAndRoles to allow Sub Projects (API families) with multiple API repositories

### LINKED ISSUE #142: New proposal how to manage "API Families" as Sub Projects with a set of API "family member" repositories
EDIT: the discussion of the proposal led to the decision to drop the "Lead_Repository" from the solution proposal, but define an API Family Sub Project simpler as a set of one or multiple API repositories. This change is not reflected here, but described within https://github.com/camaraproject/Governance/issues/142#issuecomment-2175408112

----

**Problem description**

The work within Release Management has shown that within one GitHub repository only one API or a few closely related APIs can be managed. The defined [Release Process](https://wiki.camaraproject.org/display/CAM/API+Release+Process) decouples the release numbering from the API version numbering, which allows in principle different API versions within one repository. But as a release is always on the complete repository including all APIs which are in the repository these APIs must always released at the same time, going through the same release cycle (of alpha/rc pre-releases and the public-release).

That wouldn't fit for APIs which are in different life-cycle staging, e.g. APIs which are in initial development phase with rapid releases and stable APIs which are following the meta-release cycle. As a consequence im many cases an API needs to have an own repository for to describe, develop, document and test of the API. Even if the API would fit into an existing "API Family" maintained by an existing CAMARA Sub Project.

Examples are "SimpleEdgeDiscovery", which belongs to "EdgeCloud", but has been split out into an own repository to allow proper release management, and "Tenure" which has been created in an own repository but is supposed to be maintained as part of the "KnowYourCustomer" family. Further APIs are in the API Backlog which have similar needs, or are currently together in a repository

Creating an own Sub Project for each API repository within CAMARA will not scale, as each Sub Project will need a group of (experienced) Maintainers and has administrative overhead (Wiki page, mailing list, regular meetings). A previous proposal to create a working group around a set of Sub Projects would not solve this administrative overhead but instead add another level of administrative overhead. Instead there is the need for an approach which limits the amount of Sub Projects in CAMARA but allows to develop and maintain a large number of APIs.

**Possible evolution**

The proposal is the following:

Each CAMARA Sub Project for the development of APIs will have a "Lead_Repository", which is the "home" of the API family and optional one or multiple ~~"child"~~ repositories to describe, develop, document and test API family members (name to be defined, for the purpose here: "API_Repository")

The Lead_Repository of a CAMARA API Family Sub Project will have all the components defined today in https://github.com/camaraproject/Governance/blob/main/ProjectStructureAndRoles.md for a lead repository (see also https://github.com/camaraproject/Template_Lead_Repository). It includes especially the MAINTAINERS.md file for the Sub Project, which defines the Maintainer team which is responsible for all APIs within the "family".

The Lead_Repository may contain one or few closely related APIs, which means that a new Sub Project which has (initially) only one API to develop will need only the lead repository (= which mean initially no change for most existing Sub Projects). But the Lead_Repository might have also no API definition itself, being only the home of common artefacts and documentation of the API family.

An API_Repository is linked to a Lead Repository and clearly defined (within the Readme) as a repository to describe, develop, document and test a specific API or closely related set of APIs in the context of a Sub Project API Family. The MAINTAINERS.md file points to the MAINTAINERS.md file in the Lead Repository (= the Maintainers of the Sub Project have the same responsibilities also for the API family member). The CODEOWNERS can be distinct from the Lead Repository (to distribute the workload within the Sub Project). Regarding mailing list, meeting schedule and meeting minutes there will be a link to the Lead Repository.

The approach of Lead_Repository and API_Repository allows the following "operations":

* Split out of APIs out of a Sub Project repository into a new API Repository (for release management purposes) - example "SimpleEdgeDiscovery".
* Create an API_Repository for the development of a new API (proposal) within the context of an existing Sub Project - example "Tenure" as part of "KYC" API Family (in this case the Maintainer teams need still to be merged with the new approach)
* Change existing Sub Project Lead Repositories into API_Repositories to link them together under one Lead_Repository (in this case the Maintainers are either already the same or need to agree on a (merged) group - example "OTPValidation" and "NumberVerification"
* Create an API Proposal initially as Sandbox Sub Project and decide only later if and to which API Family it can be belong ("incubating" a Sandbox project is still also an option)
* Move an API_Repository from one Sub Project to another (to restructure the API portfolio).

The sum of these operations allows to reorganise the API portfolio within CAMARA if needed, e.g. the consolidation of stable API definitions into API Family Sub Projects with larger number of APIs for long-term maintenance and evolution.

**Alternative solution**

The previous proposal described in https://github.com/camaraproject/Governance/issues/135 (one Sub Project per API, combining them via a working group) would not scale, see above in the problem description.

**Additional context**

Within the Governance documentation this approach can be defined with relative few additions, mainly the definition of an "API_Repository" (and the related Template repository).

### PR DESCRIPTION
**Pull request by hdamker**
#### What type of PR is this?

* enhancement

#### What this PR does / why we need it:

Update of ProjectStructureAndRoles.md according to the proposal in #142, specifically https://github.com/camaraproject/Governance/issues/142#issuecomment-2175408112.

Minor update of ProjectCharter with addition of definition "Repository" and reflecting that Sub Projects and Working Groups can have multiple repositories.

#### Which issue(s) this PR fixes:

Fixes #142 

#### Special notes for reviewers:

As there are major changes of the "Sub Project" chapter, the diff does not work properly. In addition some sentences are moved into the "Repositories" chapter.

#### Additional documentation 

This section can be blank.

### CODE REVIEW

**Review by bigludo7 — DISMISSED** [review]
LGTM/

**Comment by MarkusKuemmerle** [comment]
We should add also a section for IdentityAndConsentManagement, this working group is missing

**Comment by MarkusKuemmerle** [comment]
Do TSC want do decide only on Sub Projects and Working Groups or on repositories as well? Then the description of the TSC repsonsibilities in the Project Charter has to be adapted too.

File: ProjectStructureAndRoles.md
Comment (eric-murray) [inline ProjectStructureAndRoles.md:20]:
```suggestion
Some Sub Project repositories may have distinct (although sometimes overlapping) sets of Contributors, Codeowners, and Maintainers.
```

File: ProjectStructureAndRoles.md
Comment (eric-murray) [inline ProjectStructureAndRoles.md:38]: Suggested changes to improve readability
```suggestion
Only Markdown format shall be used as the document standard for documents in the main branch of Sub Project and Working Group repositories. The only exceptions are supporting documents, for which other formats like PDF or PPT are allowed. Note: Documents which are meant as base for discussion or to explain change requests are best attached to an issue. Informal supporting documents are best located within the Sub Project or Working Group wiki.
```

**Comment by hdamker** [comment]
> We should add also a section for IdentityAndConsentManagement, this working group is missing

That isn't in scope of this issue/PR - but thanks for the hint.

File: ProjectStructureAndRoles.md
Comment (hdamker) [reply ProjectStructureAndRoles.md:20]: The sentence was actually unchanged and addressed the relation between Sub Projects. For the relation between the repository *within* a Sub Project the overlap is even recommended within the content in the paragraphs above.

As this sentence here does not add any additional guidance I propose to delete it from the document.

**Comment by hdamker** [comment]
@MarkusKuemmerle 

> Do TSC want do decide only on Sub Projects and Working Groups or on repositories as well? Then the description of the TSC repsonsibilities in the Project Charter has to be adapted too.

That is already covered within the ProjectCharter, at least for the case where a new API repository does change the scope of  a Sub Project:

> * approving Sub Project proposals (including, but not limited to, incubation, deprecation, and **changes to a Sub Project’s scope**)

And the description of the API Backlog working group here in the document addresses that as well:

> The responsibility of the Working Group is to coordinate and prepare all the material needed to evaluate a proposal for a new Sub Project (consisting of one or more related APIs), or scope change of an existing Sub Project (including adding additional APIs to that Sub Project)

Adding a repository to a Sub Project for technical reasons without a scope change does not need the approval of the TSC - at least that is my view.

File: ProjectStructureAndRoles.md
Comment (eric-murray) [reply ProjectStructureAndRoles.md:20]: OK. Fine to delete it. I thought it was trying to say that the different repositories that form a Sub Project do not need to have common contributors and maintainers, though indeed that should be encouraged.

Comment (hdamker) [reply ProjectStructureAndRoles.md]: Deletion of the sentence done.

**Review by eric-murray — APPROVED** [review]

**Review by bigludo7 — APPROVED** [review]
LGTM/

**Comment by hdamker** [comment]
As decided within last TSC call we are ready to merge here.

**Pull request merged**
